Add API-Readiness-Checklist.md to RM project (#29)
camaraproject/ReleaseManagement: documentation/API-Readiness-Checklist.md

# API Readiness Checklist
| Nr | API release assets                           | alpha | release-candidate |  public-release<br>initial | public-release<br> | Status for <api-name> <version> in rX.Y |

> |----|----------------------------------------------|-------|-------------------|---------|--------|
> |  1 | API definition                               |   M   |         M         |    M    |    M   |
> |  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |
> |  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |
> |  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |
> |  5 | API documentation                            |   M   |         M         |    M    |    M   |
> |  6 | User stories                                 |   O   |         O         |    O    |    M   |
> |  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |
> |  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |
> |  9 | Test result statement                        |   O   |         O         |    O    |    M   |
> | 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |
> | 11 | Change log updated                           |   M   |         M         |    M    |    M   |
> | 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |

Reviewer: Adding the missing column and center the others:
```suggestion
|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|---------------------|
|  1 | API definition                               |   M   |         M         |    M    |    M   |                     |
|  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |                     |
|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |                     |
|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |                     |
|  5 | API documentation                            |   M   |         M         |    M    |    M   |                     |
|  6 | User stories                                 |   O   |         O         |    O    |    M   |                     |
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |                     |
|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |                     |
|  9 | Test result statement                        |   O   |         O         |    O    |    M   |                     |
| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |                     |
| 11 | Change log updated                           |   M   |         M         |    M    |    M   |                     |
| 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |                     |
```

> # API Readiness Checklist

Reviewer: ```suggestion
# API Readiness Checklist

Please find explanations for the criteria in the following checklist within [API Release Process](https://wiki.camaraproject.org/x/AgAVAQ#APIReleaseProcess-APIreadinesschecklist)
```

Reviewer: Could alternatively also go below the table.

> | Nr | API release assets                           | alpha | release-candidate |  public-|release |
> |    |                                              |       |                   | initial | stable |

Reviewer: a) I suggest to add a column where APIs can add their status or how it is intended to use the table?
b) Two lines in the header seem not to work in GitHub MarkDown, the following suggestion might work but I have not tested it.

```suggestion
| Nr | API release assets                           | alpha | release-candidate |  public-release<br>initial | public-release<br> | Status for <api-name> <version> in rX.Y |
```

Reviewer: api-name, version and release can also go in a section before the table.

|  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |      |      |
|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |      |      |
|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |      |      |
|  5 | API documentation                            |   M   |         M         |    M    |    M   |      |      |
|  6 | User stories                                 |   O   |         O         |    O    |    M   |      |      |
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |      |      |
|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |      |      |
|  9 | Test result statement                        |   O   |         O         |    O    |    M   |      |      |
| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |      |      |
| 11 | Change log updated                           |   M   |         M         |    M    |    M   |      |      |

> | 12 | Previous public-release was certified        |   O   |         O         |    O    |    M   |      |      |

Reviewer 2: If enhanced test plan is not required for the initial public release, and certification of the initial public release is required to have a stable public-release, this is assuming that the test plan used for the first certification is just the basic one, not the enhanced one, is this right? Or is that the initial public release has to be certified once the complete test plan is available.
